contracts/build.rs

```diff
@@ -74,86 +89,97 @@ fn main() {
+            .add_network_str("1", "0xC0AEe478e3658e2610c5F7A4A2E1777cE9e4f2Ac")
+            .add_network_str("4", "0xc35DADB65012eC5796536bD9864eD8773aBc74C4")
+            .add_network_str("100", "0xc35DADB65012eC5796536bD9864eD8773aBc74C4")
     });
     generate_contract_with_config("UniswapV2Router02", |builder| {
         builder
-            .add_deployment_str(1, "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D")
-            .add_deployment_str(4, "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D")
-            .add_deployment_str(100, "0x1C232F01118CB8B424793ae03F870aa7D0ac7f77")
+            .add_network_str("1", "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D")
+            .add_network_str("4", "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D")
+            .add_network_str("100", "0x1C232F01118CB8B424793ae03F870aa7D0ac7f77")
     });
     generate_contract_with_config("UniswapV2Factory", |builder| {
         builder
-            .add_deployment_str(1, "0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f")
-            .add_deployment_str(4, "0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f")
-            .add_deployment_str(100, "0xA818b4F111Ccac7AA31D0BCc0806d64F2E0737D7")
+            .add_network_str("1", "0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f")
+            .add_network_str("4", "0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f")
+            .add_network_str("100", "0xA818b4F111Ccac7AA31D0BCc0806d64F2E0737D7")
     });
     generate_contract_with_config("WETH9", |builder| {
         builder
-            .add_deployment_str(1, "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2")
-            .add_deployment_str(4, "0xc778417E063141139Fce010982780140Aa0cD5Ab")
-            .add_deployment_str(100, "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d")
+            .add_network_str("1", "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2")
+            .add_network_str("4", "0xc778417E063141139Fce010982780140Aa0cD5Ab")
+            .add_network_str("100", "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d")
     });
 }
 
 fn generate_contract(name: &str) {
     generate_contract_with_config(name, |builder| builder)
 }
 
-fn generate_contract_with_config(name: &str, config: impl FnOnce(Builder) -> Builder) {
-    let artifact = paths::contract_artifacts_dir()
+fn generate_contract_with_config(
+    name: &str,
+    config: impl FnOnce(ContractBuilder) -> ContractBuilder,
+) {
+    let path = paths::contract_artifacts_dir()
         .join(name)
         .with_extension("json");
+    let contract = TruffleLoader::new()
+        .name(name)
+        .load_contract_from_file(&path)
+        .unwrap();
     let address_file = paths::contract_address_file(name);
     let dest = env::var("OUT_DIR").unwrap();
 
-    println!("cargo:rerun-if-changed={}", artifact.display());
-    let mut builder = Builder::new(artifact)
-        .with_contract_name_override(Some(name))
-        .with_visibility_modifier(Some("pub"));
+    println!("cargo:rerun-if-changed={}", path.display());
```

Comment:
This is what I was looking for in https://github.com/gnosis/ethcontract-rs/issues/540! Cool!